File: fruitable-server/src/main/java/com/fruitable/Service/ReportService.java
package com.fruitable.Service;


import com.fruitable.fileResponse.ReportPageableResponse;
import com.fruitable.model.Report;

public interface ReportService {

	// add report
	public Report addReport(Report report);
	
	// update report
	public Report updateReport(Report repot);
	
	// get all report
	public ReportPageableResponse getAllReport(Integer pageNumber, Integer pageSize, String sortBy); 
	
	
}

File: fruitable-front/src/app/service/report.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ReportService {

  constructor(
    private http: HttpClient
  ) { }

  // add report
  public addReport(report:any){
    return this.http.post(`${baseUrl}/report/`,report);
  }

  // get all report
  public getAllReport(pageNumber?:number,pageSize?:number){
    if(pageNumber != null && pageSize != null){
      return this.http.get(`${baseUrl}/report/?pageNumber=${pageNumber}&pageSize=${pageSize}`);
    }else{
      return this.http.get(`${baseUrl}/report/`);
    }
  }

}
